runtime/gc_modron_startup/mminit.cpp

```diff
@@ -3196,7 +3199,86 @@ gcReinitializeDefaultsForRestore(J9VMThread* vmThread)
 		extensions->gcThreadCountForced = false;
 	}
 
-	return result;
+	/* While ideally it is desired to change the heap geometry (parameters such as
+	 * extensions->maxMemory, virtual memory reserved, etc), it would be too complex
+	 * and slow. As an alternative, the geometry remains unchanged, but the
+	 * existing softmx functionality is utilized to limit the maximum size of heap.
+	 */
+	{
+		/* Softmx is initially set by heuristic(a function of usable physical memory),
+		 * and then further adjusted or refused based on -Xmx/-Xms
+		 */
+		uintptr_t candidateSoftMx = 0;
+		if (0.0 <= extensions->maxRAMPercent) {
```

Comment:
Following current logic maxRAMPercent should be taking to consideration only in the case if -Xmx is not specified.